src/Arcus.WebApi.Security/Authentication/Certificates/CertificateAuthenticationFilter.cs

```diff
@@ -33,71 +38,95 @@ public async Task OnAuthorizationAsync(AuthorizationFilterContext context)
+                    context.Result = new UnauthorizedObjectResult("Client certificate in request is not allowed");
                 }
             }
+            else
+            {
+                LogSecurityEvent(logger, LogLevel.Error, "No client certificate is specified in the request while this authentication filter requires a certificate to validate on the configured validation requirements", HttpStatusCode.Unauthorized);
+                context.Result = new UnauthorizedObjectResult("No client certificate found in request");
+            }
         }
 
-        private static X509Certificate2 GetOrLoadClientCertificateFromRequest(HttpContext context, ILogger logger)
+        private static bool TryGetClientCertificateFromRequest(HttpContext context, ILogger logger, out X509Certificate2 clientCertificate)
         {
-            if (context.Connection.ClientCertificate is null)
+            if (context.Connection.ClientCertificate != null)
+            {
+                clientCertificate = context.Connection.ClientCertificate;
+                return clientCertificate != null;
+            }
+
+            if (!context.Request.Headers.TryGetValue(HeaderName, out StringValues headerValues))
             {
-                const string headerName = "X-ARR-ClientCert";
+                logger.LogError("Cannot load client certificate because request header {HeaderName} was not found", HeaderName);
 
-                try
-                {
-                    if (context.Request.Headers.TryGetValue(headerName, out StringValues headerValue))
-                    {
-                        byte[] rawData = Convert.FromBase64String(headerValue);
-                        return new X509Certificate2(rawData);
-                    }
-                }
-                catch (Exception exception)
+                clientCertificate = null;
+                return false;
+            }
+
+            try
+            {
+                var headerValue = headerValues.ToString();
+                if (!String.IsNullOrWhiteSpace(headerValue) 
+                    && headerValue.Trim().Length % 4 == 0 
+                    && Base64Regex.IsMatch(headerValue))
                 {
-                    logger.LogError(exception, "Cannot load client certificate from {headerName} header", headerName);
+                    byte[] rawData = Convert.FromBase64String(headerValue);
+                    clientCertificate = new X509Certificate2(rawData);
+                    return true;
                 }
+
+                logger.LogError(
+                    "Cannot load client certificate from request header {HeaderName} because the header value is not a valid base64 encoded string",
+                    HeaderName);
+            }
+            catch (Exception exception)
+            {
+                logger.LogError(exception, "Cannot load client certificate from {HeaderName} header due to an unexpected exception", HeaderName);
```

Comment:
But here an error here is fine for example

Reply:
Yes, I see. Here there's something wrong with the certificate itself. Unexpected.

Reply:
I'll resolve this.